tensorboard/webapp/metrics/views/main_view/common_selectors_test.ts

```diff
@@ -430,4 +552,351 @@ describe('common selectors', () => {
+      );
+      expect(exp1Result.length).toEqual(2);
+      expect(exp1Result[0].run).toEqual({...run1, experimentId: 'exp1'});
+      expect(exp1Result[1].run).toEqual({...run2, experimentId: 'exp1'});
+
+      const exp2Result = selectors.TEST_ONLY.utils.getRenderableRuns(['exp2'])(
+        state
+      );
+      expect(exp2Result.length).toEqual(3);
+      expect(exp2Result[0].run).toEqual({...run2, experimentId: 'exp2'});
+      expect(exp2Result[1].run).toEqual({...run3, experimentId: 'exp2'});
+      expect(exp2Result[2].run).toEqual({...run4, experimentId: 'exp2'});
+    });
+
+    it('returns two runs when a run is associated with multiple experiments', () => {
+      const result = selectors.TEST_ONLY.utils.getRenderableRuns([
+        'exp1',
+        'exp2',
+      ])(state);
+      expect(result.length).toEqual(5);
+      expect(result[0].run).toEqual({...run1, experimentId: 'exp1'});
+      expect(result[1].run).toEqual({...run2, experimentId: 'exp1'});
+      expect(result[2].run).toEqual({...run2, experimentId: 'exp2'});
+      expect(result[3].run).toEqual({...run3, experimentId: 'exp2'});
+      expect(result[4].run).toEqual({...run4, experimentId: 'exp2'});
+    });
+
+    it('returns empty list when no experiments are provided', () => {
+      expect(selectors.TEST_ONLY.utils.getRenderableRuns([])(state)).toEqual(
+        []
+      );
+    });
+  });
+
+  describe('filterRunItemsByHparamAndMetricFilter', () => {
+    it('filters by hparams using discrete filters', () => {
+      expect(
+        selectors.TEST_ONLY.utils.filterRunItemsByHparamAndMetricFilter(
+          runTableItems,
+          new Map([
+            [
+              'lr',
+              {
+                type: DomainType.DISCRETE,
+                includeUndefined: false,
+                possibleValues: [1, 3, 5],
+                filterValues: [1],
+              },
+            ],
+          ]),
+          new Map()
+        ).length
+      ).toEqual(1);
+      expect(
+        selectors.TEST_ONLY.utils.filterRunItemsByHparamAndMetricFilter(
+          runTableItems,
+          new Map([
+            [
+              'lr',
+              {
+                type: DomainType.DISCRETE,
+                includeUndefined: false,
+                possibleValues: [1, 3, 5],
+                filterValues: [1, 5],
+              },
+            ],
+          ]),
+          new Map()
+        ).length
+      ).toEqual(2);
+
+      expect(
+        selectors.TEST_ONLY.utils.filterRunItemsByHparamAndMetricFilter(
+          runTableItems,
+          new Map([
+            [
+              'who knows',
+              {
+                type: DomainType.DISCRETE,
+                includeUndefined: false,
+                possibleValues: [1, 3, 5],
+                filterValues: [1, 5],
+              },
+            ],
+          ]),
+          new Map()
+        ).length
+      ).toEqual(0);
+    });
+
+    it('filters by hparams using interval filters', () => {
```

Comment:
Not sure we need to test the different types of hparam or metric filters as that is already tested in the matchfilter test. I am ok with leaving them in though.